libcontainer/standard_init_linux.go

```diff
@@ -187,7 +185,16 @@ func (l *linuxStandardInit) Init() error {
 			return newSystemErrorWithCause(err, "init seccomp")
 		}
 	}
-	if err := syscall.Exec(name, l.config.Args[0:], os.Environ()); err != nil {
+	runtime.LockOSThread()
```

Comment:
Did you try moving `runtime.LockOSThread()` up higher instead of moving the selinux code down?  I'm worried about seccomp and selinux in this flow.

Reply:
I agree, and would just put `runtime.LockOSThread` to the very beginning of `init` for both files and keep the rest of the code the same as before.

Reply:
The issue that triggered this PR was this AVC.
```
podman run -ti --blkio-weight-device /dev/sda1:500 alpine /bin/sh

triggered

type=AVC msg=audit(1528390506.883:851): avc:  denied  { append } for  pid=6511 comm="sh" name="blkio.weight_device" dev="cgroup" ino=56622 scontext=system_u:system_r:container_t:s0:c203,c914 tcontext=system_u:object_r:cgroup_t:s0 tclass=file
```
This avc shows a that runc launched a comm="sh" that is attempting to append to the blkio.weight_device file in the cgroup directory.  The problem is the shell is running as the container label, container_t.  Since this is part of the setup it should not be confined by the container type.  
It is my theory that runc launched a process after the SELinux label was set, which ran with the container label.  Moving the SetProcessLabel down in the stack means this will not happen.  I hav e no problem locking the threads much higher, if that is what you want.

Reply:
I think it would be better to lock sooner before changing where selinux is being set.

Reply:
I will lock at the top, but that does not eliminate the risk over where SELinux is set.